pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -115,20 +104,29 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	// update the validations in pkg/webhook.validateContainer.
 	userContainer.Name = UserContainerName
 	userContainer.Resources = userResources
-	userContainer.Ports = userPorts
 	userContainer.VolumeMounts = append(userContainer.VolumeMounts, varLogVolumeMount)
 	userContainer.Lifecycle = userLifecycle
-	userContainer.Env = append(userContainer.Env, userEnv)
+
+	userPort, bIsFind := getUserPort(userContainer.Ports)
+	if !bIsFind {
+		userPort = &corev1.ContainerPort{
+			Name:          userPortName,
+			ContainerPort: int32(defaultUserPort),
+		}
+		userContainer.Ports = append(userContainer.Ports, *userPort)
+	}
+	userPortEnv := getUserPortEnv(userPort)
```

Comment:
I would rename "getUserPortEnv" to "buildUserPortEnv"

Reply:
ok, I get it